File: languages/java/tests/src/test/java/co/revaly/sdk/ConnectTimeoutDefaultTests.java
package co.revaly.sdk;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.time.Duration;
import org.junit.jupiter.api.Test;

/**
 * ADR-SDK-029 connect-timeout-default semantics: unset resolves to the 10-second edge-ratified
 * default, noConnectTimeout() opts out entirely, explicit values pass through, and zero/negative
 * values are rejected at build.
 */
class ConnectTimeoutDefaultTests {

    @Test
    void unsetResolvesToTheRatifiedDefault() {
        assertEquals(Duration.ofSeconds(10), RapClient.DEFAULT_CONNECT_TIMEOUT);
        assertEquals(
                RapClient.DEFAULT_CONNECT_TIMEOUT, RapClient.builder().effectiveConnectTimeout());
    }

    @Test
    void noConnectTimeoutOptsOutOfAnySdkConnectBound() {
        assertNull(RapClient.builder().noConnectTimeout().effectiveConnectTimeout());
        assertNull(RapClient.builder().connectTimeout(null).effectiveConnectTimeout());
    }

    @Test
    void explicitValuesPassThroughUnchanged() {
        RapClient.Builder builder = RapClient.builder().connectTimeout(Duration.ofSeconds(3));
        assertEquals(Duration.ofSeconds(3), builder.effectiveConnectTimeout());
    }

    @Test
    void zeroAndNegativeConnectTimeoutsAreRejectedAtBuild() {
        RapClient.Builder zero = RapClient.builder().apiKey("sk-synthetic-test");
        zero.connectTimeout(Duration.ZERO);
        assertThrows(IllegalArgumentException.class, zero::build);

        RapClient.Builder negative = RapClient.builder().apiKey("sk-synthetic-test");
        negative.connectTimeout(Duration.ofSeconds(-5));
        assertThrows(IllegalArgumentException.class, negative::build);
    }
}

File: languages/dotnet/tests/Revaly.Sdk.Tests/ConnectTimeoutDefaultTests.cs
using Revaly.Sdk.Testing;

namespace Revaly.Sdk.Tests;

/// <summary>
/// ADR-SDK-029 connect-timeout-default semantics: unset resolves to the 10-second
/// edge-ratified default, Timeout.InfiniteTimeSpan opts out of any SDK connect bound,
/// explicit values pass through, and zero/negative values are rejected at construction.
/// </summary>
public class ConnectTimeoutDefaultTests
{
    private static RapClientOptions Options(TimeSpan? connectTimeout = null) => new()
    {
        ApiKey = SyntheticData.TestApiKey,
        BaseUrl = new Uri("https://sandbox.synthetic.test"),
        Transport = new RapMockTransport(),
        ConnectTimeout = connectTimeout,
    };

    [Fact]
    public void Unset_resolves_to_the_ratified_default()
    {
        Assert.Equal(TimeSpan.FromSeconds(10), RapClientOptions.DefaultConnectTimeout);
        Assert.Equal(
            RapClientOptions.DefaultConnectTimeout,
            Options().EffectiveConnectTimeout);
    }

    [Fact]
    public void InfiniteTimeSpan_opts_out_of_any_sdk_connect_bound()
    {
        Assert.Null(Options(System.Threading.Timeout.InfiniteTimeSpan).EffectiveConnectTimeout);
    }

    [Fact]
    public void Explicit_values_pass_through_unchanged()
    {
        Assert.Equal(
            TimeSpan.FromSeconds(3),
            Options(TimeSpan.FromSeconds(3)).EffectiveConnectTimeout);
    }

    [Theory]
    [InlineData(0)]
    [InlineData(-5)]
    public void Zero_and_negative_connect_timeouts_are_rejected_at_construction(int seconds)
    {
        var ex = Assert.Throws<ArgumentException>(
            () => new RapClient(Options(TimeSpan.FromSeconds(seconds))));
        Assert.Equal(nameof(RapClientOptions.ConnectTimeout), ex.ParamName);
    }
}
